### PR TITLE
stage 1 - added description and changed label to name

### PR DESCRIPTION
**Pull request by julieheard**
I am expanding the globalConfig plugin. I have renamed label to name and added a description textbox

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,13 +27,7 @@
         <email>bhacker@nowhere.net</email>
       </developer>
     </developers> -->
-
-    <scm child.scm.connection.inherit.append.path="false" child.scm.developerConnection.inherit.append.path="false" child.scm.url.inherit.append.path="false">
-        <connection>scm:git:https://github.com/${gitHubRepo}</connection>
-        <developerConnection>scm:git:https://github.com/${gitHubRepo}</developerConnection>
-        <tag>${scmTag}</tag>
-        <url>https://github.com/${gitHubRepo}</url>
-    </scm>    <properties>
```
Comment (jtnord) [inline pom.xml:36]: any particular reason this was removed?

not that you would need it bacause you most likely will never be be releasing this, but it does not harm things to have it here, plus it is also used to put information into the [plugins manifiest](https://github.com/jenkinsci/maven-hpi-plugin/blob/533f4d29d23b9e69c223d3897bed467b243a0300/src/main/java/org/jenkinsci/maven/plugins/hpi/AbstractJenkinsManifestMojo.java#L216).

File: src/main/java/io/jenkins/plugins/sample/SampleConfiguration.java [hunk 2]
```diff
@@ -19,31 +22,49 @@ public static SampleConfiguration get() {
...
+            //Check format of name (letters and hyphens only)
+            Pattern pattern = Pattern.compile("^[A-Za-z\\s-]+$");
+            Matcher matcher = pattern.matcher(value);
+            if(!matcher.matches()){
+                setName("");
+                return FormValidation.warning("Please enter a valid name (letters and hyphens only).");
+            }
         }
         return FormValidation.ok();
     }
```
Comment (jtnord) [inline src/main/java/io/jenkins/plugins/sample/SampleConfiguration.java:70]: this would be stage  2...

**Review by jtnord — COMMENTED** [review]
Looks ok - contains parts of Stage2 though

File: pom.xml (hunk 1 above)
Comment (julieheard) [reply pom.xml:36]: Only removed as am getting error 'Attribute child.scm.connection.inherit.append.path is not allowed here'

**Pull request merged**